src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs

```diff
@@ -839,31 +842,59 @@ private async Task<ExtendLedgerReport> ProcessTransactions(ReadWriteDbContext db
+                                });
+                            }
+                            else if (nonFungibleIds?.Any() == true)
+                            {
+                                nonFungibleWithdrawalTransactionEvents.Add(new NonFungibleWithdrawalTransactionEvent(globalAncestorId, resourceEntityId, nonFungibleIds, stateVersion));
+                            }
+                            else
+                            {
+                                throw new InvalidOperationException("Unable to process data_json property");
+                            }
                         }
 
                         // TODO "deposit" event
                         if (@event.Type.LocalTypeIndex.Index == 28)
                         {
                             var globalAncestorId = eventEmitterEntity.DatabaseGlobalAncestorId;
                             var resourceEntityId = eventEmitterEntity.GetDatabaseEntity<VaultEntity>().ResourceEntityId;
+                            var data = (JObject)@event.Data.DataJson;
+                            var fungibleAmount = data["fields"]?[0]?["value"]?.ToString();
+                            var nonFungibleIds = data["fields"]?[0]?["elements"]?.Select(x => x.ToString()).ToList();
 
-                            ledgerTransactionSearchIndicesToAdd.Add(new LedgerTransactionSearchIndex
+                            if (fungibleAmount != null)
+                            {
+                                ledgerTransactionEventsToAdd.Add(new DepositFungibleResourceLedgerTransactionEvent
+                                {
+                                    Id = sequences.LedgerTransactionEventSequence++,
+                                    TransactionStateVersion = stateVersion,
+                                    EntityId = globalAncestorId,
+                                    ResourceEntityId = resourceEntityId,
+                                    Amount = TokenAmount.FromDecimalString(fungibleAmount),
+                                });
+                            }
+                            else if (nonFungibleIds?.Any() == true)
+                            {
+                                nonFungibleDepositTransactionEvents.Add(new NonFungibleDepositTransactionEvent(globalAncestorId, resourceEntityId, nonFungibleIds, stateVersion));
+                            }
+                            else
                             {
-                                Id = sequences.LedgerTransactionSearchIndexSequence++,
-                                TransactionStateVersion = stateVersion,
-                                EntityId = globalAncestorId,
-                                OperationType = LedgerTransactionSearchIndexOperationType.Deposit,
-                                OperationResourceEntityId = resourceEntityId,
-                            });
+                                throw new InvalidOperationException("Unable to process data_json property");
```

Comment:
I'm not sure that this message tells what is happening here, can we change it to sthn like "Unexpected structure of deposit event, unable to parse" ?